Source/OCMockito/Core/MKTMockingProgress.m

```diff
@@ -49,16 +51,20 @@ - (MKTOngoingStubbing *)pullOngoingStubbing
     return result;
 }
 
-- (void)verificationStarted:(id <MKTVerificationMode>)mode atLocation:(MKTTestLocation)location
+- (void)verificationStarted:(id <MKTVerificationMode>)mode atLocation:(MKTTestLocation)location withMock:(MKTBaseMockObject *)mock
 {
     self.verificationMode = mode;
+    self.verificationModeMockObject = mock;
     [self setTestLocation:location];
 }
 
-- (id <MKTVerificationMode>)pullVerificationMode
+- (id <MKTVerificationMode>)pullVerificationModeWithMock:(MKTBaseMockObject *)mock
 {
     id <MKTVerificationMode> result = self.verificationMode;
-    self.verificationMode = nil;
+
+    if ([mock isEqual:self.verificationModeMockObject])
```

Comment:
Here I only `nil` the `verificationMode` when it's the correct `mock`

Reply:
Nice!